src/Microsoft.Extensions.Configuration.AzureAppConfiguration/AzureAppConfigurationProvider.cs

```diff
@@ -547,41 +545,130 @@ private void SetDirty(TimeSpan? maxDelay)
             return applicationData;
         }
 
-        private async Task InitializeAsync(bool ignoreFailures, IEnumerable<ConfigurationClient> availableClients, CancellationToken cancellationToken = default)
+        private async Task LoadAsync(bool ignoreFailures, CancellationToken cancellationToken)
         {
-            Dictionary<string, ConfigurationSetting> data = null;
-            Dictionary<KeyValueIdentifier, ConfigurationSetting> watchedSettings = null;
-            
+            var startupStopwatch = Stopwatch.StartNew();
+
+            int exponentialBackoffAttempts = 0;
+
+            var startupExceptions = new List<Exception>();
+
+            IEnumerable<ConfigurationClient> clients = _configClientManager.GetAllClients();
+
             try
             {
-                await ExecuteWithFailOverPolicyAsync(
-                    availableClients,
-                    async (client) =>
+                while (true)
+                {
+                    if (await TryInitializeAsync(clients, startupExceptions, cancellationToken).ConfigureAwait(false))
+                    {
+                        break;
+                    }
+
+                    try
+                    {
+                        if (startupStopwatch.Elapsed < FailOverConstants.StartupFixedBackoffDuration)
+                        {
+                            await Task.Delay(startupStopwatch.Elapsed.CalculateFixedStartupBackoffDuration(), cancellationToken).ConfigureAwait(false);
+                        }
+                        else
+                        {
+                            exponentialBackoffAttempts++;
+
+                            await Task.Delay(FailOverConstants.MinBackoffDuration.CalculateExponentialStartupBackoffDuration(
```

Comment:
I'd like to see the delay calculated outside the try, and a single place where we invoke Task.Delay

Reply:
Didn't see this initially, will move it out